lib/Doctrine/ORM/Query/Expr/Comparison.php

```diff
@@ -4,51 +4,47 @@
 
 namespace Doctrine\ORM\Query\Expr;
 
+use Stringable;
+
 /**
  * Expression class for DQL comparison expressions.
  *
  * @link    www.doctrine-project.org
  */
-class Comparison
+class Comparison implements Stringable
 {
-    public const EQ  = '=';
-    public const NEQ = '<>';
-    public const LT  = '<';
-    public const LTE = '<=';
-    public const GT  = '>';
-    public const GTE = '>=';
+    final public const EQ  = '=';
+    final public const NEQ = '<>';
+    final public const LT  = '<';
+    final public const LTE = '<=';
+    final public const GT  = '>';
+    final public const GTE = '>=';
 
     /**
      * Creates a comparison expression with the given arguments.
      *
-     * @param mixed  $leftExpr
      * @param string $operator
-     * @param mixed  $rightExpr
      */
-    public function __construct(protected $leftExpr, protected $operator, protected $rightExpr)
+    public function __construct(protected mixed $leftExpr, protected $operator, protected mixed $rightExpr)
```

Comment:
If `$operator` is string, why no type hint here?

Reply:
Not sure why I left that one out, let's use a type hint.